packages/kit/src/core/sync/create_manifest_data/index.js

```diff
@@ -229,151 +272,169 @@ export default function create_manifest_data({
 
 	return {
 		assets,
-		layout,
-		error,
 		components,
 		routes,
 		matchers
 	};
 }
 
 /**
- * @param {string} needle
- * @param {string} haystack
+ * @param {string} file
+ * @param {Tree} tree
+ * @param {string[]} extensions
  */
-function count_occurrences(needle, haystack) {
-	let count = 0;
-	for (let i = 0; i < haystack.length; i += 1) {
-		if (haystack[i] === needle) count += 1;
+function trace(file, tree, extensions) {
+	/** @type {Array<string | undefined>} */
+	const layouts = [];
+
+	/** @type {Array<string | undefined>} */
+	const errors = [];
+
+	const parts = file.split('/');
+	const filename = /** @type {string} */ (parts.pop());
+	const extension = /** @type {string} */ (extensions.find((ext) => file.endsWith(ext)));
+	const base = filename.slice(0, -extension.length);
+
+	let layout_id = base.includes('@') ? base.split('@')[1] : DEFAULT;
+
+	// walk up the tree, find which __layout and __error components
+	// apply to this page
+	while (true) {
```

Comment:
looks like you need to update this line to satisfy eslint
```suggestion
	while (parts.length > 0) {
```

Reply:
Or, alternatively, something like this:

```suggestion
	for (;;) {
```

(or add an `eslint-ignore` comment).

Reply:
`parts.length > 0` isn't correct, because it will fail for files in the root directory. i added an eslint-ignore comment